modules/shortcodes.php

```diff
@@ -118,12 +118,51 @@ function jetpack_preg_replace_callback_outside_tags( $pattern, $callback, $conte
 	
 	$textarr = wp_html_split( $content );
 	unset( $content );
+
+	// Flag to check we're inside an HTML tag.
+	$inside_tag = '';
+
 	foreach( $textarr as &$element ) {
-	    if ( '' === $element || '<' === $element{0} )
+
+		// If it's an empty element, continue.
+	    if ( '' === $element ) {
 	        continue;
-	    $element = preg_replace_callback( $pattern, $callback, $element );
+		}
+
+		// If this is an opening HTML tag, save it and move to the next element.
+		if ( '<' === $element[0] && '/>' !== substr( $element, -2 ) ) {
+			$inside_tag = $element;
+			continue;
+		}
+
+		// If this is a closing HTML tag, erase the previously saved element.
+		if ( '</' === substr( $element, 0, 2 ) ) {
+			$inside_tag = '';
+			continue;
+		}
+
+		// We're inside an HTML tag.
+		if ( ! empty( $inside_tag ) ) {
+
+			// If the callback it's not in the blacklist, run it.
+			switch ( $callback ) {
+				case 'vimeo_link_callback':
+					// If we're inside a link, don't embed the video.
+					// Covers cases like "Here's a link to the video: <a href="https://vimeo.com/123456">https://vimeo.com/123456</a>".
+					if ( ! preg_match( '/<a\s[^>]+>/i', $inside_tag ) ) {
```

Comment:
If `wp_html_split()` truly does it's job, then all we need to do here is verify that this isn't an anchor tag, right? then maybe if that's true, we can both simplify the `RegEx` _and_ save ourselves headaches from cases where it incorrectly matches.

```php
// are we inside an anchor tag?
if ( 1 !== preg_match( '/^<a\s+/i', $inside_tag ) && '>' !== substr( $inside_tag, -1 ) ) {
	$element = …;
}
```